File: frontend/src/lib/api/schemas/audit.ts
import { z } from "zod";

export const auditLogReadSchema = z.object({
  id: z.string().uuid(),
  actor_user_id: z.string().uuid().nullable(),
  action: z.string(),
  entity_type: z.string(),
  entity_id: z.string().uuid(),
  before: z.record(z.string(), z.unknown()).nullable(),
  after: z.record(z.string(), z.unknown()).nullable(),
  ip_address: z.string().nullable(),
});
export type AuditLogRead = z.infer<typeof auditLogReadSchema>;

File: backend/src/ytforge/interfaces/api/schemas/audit.py
from __future__ import annotations

import uuid
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    actor_user_id: uuid.UUID | None
    action: str
    entity_type: str
    entity_id: uuid.UUID
    before: dict[str, Any] | None
    after: dict[str, Any] | None
    ip_address: str | None
